src/consensus/vote.cpp

```diff
@@ -549,6 +564,106 @@ std::string VoteManager::getJsonStr(std::vector<Vote> const& votes) {
   return ptroot.toStyledString();
 }
 
+std::vector<Vote> VoteManager::getProposalVotes(uint64_t const pbft_round) {
+  std::unordered_map<blk_hash_t, std::unordered_set<vote_hash_t>> proposal_voted_value_map;
+  {
+    sharedLock_ lock(verified_votes_access_);
+    proposal_voted_value_map = verified_votes_[pbft_round][1];
```

Comment:
why would you create a copy of whole std::unordered_map<blk_hash_t, std::unordered_set<vote_hash_t>> ? Is that because you did not want to lock verified_votes_access_ for the whole processing functionality below ? Because this might a lot of memory to copy

Reply:
Yeah, because don't want to lock verified_votes_access_.
This will not a lot of memory to copy. Since only allow to vote one single vote in proposal phase per each node. All 3 boot nodes occupied 2t+1 votes, 3 boot nodes only vote 3 votes at proposal phase. And there are only t-1 nodes will get sortition in maximum. So the max number of proposal votes should be t - 1 + 3 = t + 2 = 1/3 committee + 2 = 335
Even after remove centralized by boot nodes, the max number of proposal votes for each round is committee size, that is 1000. So this is not a lot of memory

Reply:
Since moved back vote struct in memory, don't need retrieve vote from DB. Remove copy table